In python,
 input() is basic input operator
 print() is the basic output operator

input()

Reading a string input from user

In [2]:
name = input("Enter your name: ")
print("Hello, " + name + "!")

Hello, b!


Reading integer input from user

In [3]:
age = int(input("Enter your age: "))
print("You will be " + str(age + 1) + " years old next year.")

ValueError: invalid literal for int() with base 10: 'df d'

reading multiple inputs from user

In [4]:
name, age = input("Enter your name and age, separated by a space: ").split()  #split() method splits(using space character here) the string into a list
print("Hello, " + name + "! You are " + age + " years old.")

Hello, cc! You are 3 years old.


using a loop to read multiple inputs from user

In [ ]:
numbers = []
for i in range(5):  #read 5 numbers from the user
    num = int(input("Enter a number: "))
    numbers.append(num)
print("The sum of the numbers is:", sum(numbers))

printing a string

In [ ]:
print("Hello, world!")

printing a variable

In [ ]:
name = "Alice"
print("Hello, " + name + "!")

printing multiple variables

In [9]:
name = "Bob"
age = 25
print( name, age,"years old.")
name = "Bob"
age = 25
print("Hello, " + name + "! You are " + str(age) + " years old.")

Bob 25 years old.
Hello, Bob! You are 25 years old.


printing with format()

In [5]:
name = "Charlie"
age = 30
print("Hello, {}! You are {} years old.".format(name, age))

Hello, Charlie! You are 30 years old.



Here are some other usages of the `format()` method in Python:

1. Specifying the order of arguments:


In [6]:
name = "Alice"
age = 25
print("{1} is {0} years old.".format(age, name))

Alice is 25 years old.


This will output the string "Alice is 25 years old." to the console, by specifying the order of the arguments in the `format()` method.

2. Formatting numbers:


In [7]:
x = 123.456
print("The value of x is {:.2f}".format(x))

The value of x is 123.46


This will output the string "The value of x is 123.46" to the console, by formatting the number `x` to two decimal places using the `:.2f` format specifier.

3. Formatting strings:


In [8]:
name = "Bob"
print("Hello, {:_<10}!".format(name))

Hello, Bob_______!


This will output the string "Hello, Bob_______!" to the console, by formatting the string `name` to a width of 10 characters and filling the remaining space with underscores using the `:_<10` format specifier.

4. Formatting dates and times:


In [9]:
import datetime
now = datetime.datetime.now()
print("The current time is {:%Y-%m-%d %H:%M:%S}".format(now))

The current time is 2023-07-17 13:14:51


This will output the current date and time in the format "YYYY-MM-DD HH:MM:SS" to the console, using the `:%Y-%m-%d %H:%M:%S` format specifier.

In each of these examples, the `format()` method is used to format a string with one or more values, using various format specifiers to control the formatting of the values.